### ISSUE TITLE
Data updates should be rerunnable

### ISSUE THREAD

**Issue by rhymes** [forem/forem#6175]
**Describe the bug**

`rails data_updates:run` ran twice doesn't re-run the script, or at least it doesn't register it as a success the second time.

The script that was pull from master is `lib/data_update_scripts/20200217215802_index_classified_listings_to_elasticsearch.rb`

```shell
➜  devto git:(master) rails data_updates:run
2020-02-19 17:03:40 +0100: PUT http://localhost:9200/classified_listings_development_alias/_doc/1 [status:403, request:0.518s, query:N/A]
2020-02-19 17:03:40 +0100: > {"id":1,"body_markdown":"`Commodi.`","bumped_at":"2020-02-13T16:20:32.495Z","category":"cfp","contact_via_connect":true,"expires_at":null,"location":"West Terrellland","processed_html":"<p><code>Commodi.</code></p>\n","slug":"ratione-dolor-nemo-ipsa-4b70","title":"Ratione dolor nemo ipsa.","user_id":46,"tags":[],"author":{"username":"ms_eustolia_aut_goldner","name":"Ms. Eustolia Goldner Aut","profile_image_90":"/uploads/user/profile_image/46/e2a050a7-6da4-4f6a-b3bc-874c7ce43ec4.png"}}
2020-02-19 17:03:40 +0100: < {"error":{"root_cause":[{"type":"cluster_block_exception","reason":"index [classified_listings_development] blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"}],"type":"cluster_block_exception","reason":"index [classified_listings_development] blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"},"status":403}
2020-02-19 17:03:40 +0100: [403] {"error":{"root_cause":[{"type":"cluster_block_exception","reason":"index [classified_listings_development] blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"}],"type":"cluster_block_exception","reason":"index [classified_listings_development] blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"},"status":403}
➜  devto git:(master) psql PracticalDeveloper_development
PracticalDeveloper_development=# select * from data_update_scripts;
┌────┬───────────────────────────────────────────────────────────┬────────┬────────────────────────────┬────────────────────────────┬────────────────────────────┬────────────────────────────┐
│ id │                         file_name                         │ status │           run_at           │        finished_at         │         created_at         │         updated_at         │
├────┼───────────────────────────────────────────────────────────┼────────┼────────────────────────────┼────────────────────────────┼────────────────────────────┼────────────────────────────┤
│  4 │ 20200214171607_index_tags_to_elasticsearch                │      3 │ 2020-02-15 09:38:34.768666 │ 2020-02-15 09:38:35.415858 │ 2020-02-15 09:38:34.757657 │ 2020-02-15 09:38:35.421385 │
│  5 │ 20200217131245_re_index_existing_articles_with_approved   │      2 │ 2020-02-17 16:30:15.012304 │ 2020-02-17 16:30:16.527559 │ 2020-02-17 16:29:53.513183 │ 2020-02-17 16:30:16.529633 │
│  6 │ 20200217215802_index_classified_listings_to_elasticsearch │      3 │ 2020-02-19 16:03:38.248933 │ 2020-02-19 16:03:40.254466 │ 2020-02-19 16:03:38.215601 │ 2020-02-19 16:03:40.261021 │
└────┴───────────────────────────────────────────────────────────┴────────┴────────────────────────────┴────────────────────────────┴────────────────────────────┴────────────────────────────┘
```

though the errors I have are likely related to https://github.com/thepracticaldev/dev.to/issues/6085

I think this behavior is indicative of another issue.

I tried to run the script again but with no output, I guess it's ignored by the runner, but how can I re-run it? If you check my DB I have 2 scripts that have failed. Not sure about the state of my DB 🗡

**To Reproduce**

1. have less disk space than ES wants (I have 12 GB free)
2. run a script with `rails data_update_scripts:run`
3. get the error
4. run the script again, nothing ahhapens

**Expected behavior**

I expect the script to be re-executable if it has failed

**Comment by citizen428** [comment]
Looking at the code we only run `enqueued` updates, so if you change the status of the relevant update scripts from 3 to 0 you should be able to run them. 

Maybe we should add a rake task like `rails data_updates:retry` that takes an update script name as argument.

**Comment by cmgorton** [comment]
Looks like this may have been addressed with PR #12424 and #12568. Also referenced in a larger potential rfc in #6332. I am going to go a head and close this issue. Let me know if you think we should reopen it!
